nucleus/modelci/__init__.py

```diff
@@ -19,33 +32,38 @@ class ModelCI:
     """Model CI Python Client extension."""
 
     def __init__(self, api_key: str, endpoint: str):
-        self._connection = Connection(api_key, endpoint)
+        self.connection = Connection(api_key, endpoint)
 
     def __repr__(self):
-        return f"ModelCI(connection='{self._connection}')"
+        return f"ModelCI(connection='{self.connection}')"
 
     def __eq__(self, other):
-        return self._connection == other.connection
+        return self.connection == other.connection
 
-    def list_eval_functions(self) -> List[EvalFunction]:
-        """Lists all available evaluation functions. ::
+    @property
+    def eval_functions(self) -> AvailableEvalFunctions:
+        """List all available evaluation functions. ::
 
         import nucleus
         client = nucleus.NucleusClient("YOUR_SCALE_API_KEY")
 
-        eval_functions = client.modelci.list_eval_functions()
+        eval_functions = client.modelci.eval_functions
+
+        Returns:
+            :class:`AvailableEvalFunctions`: A container for all the available eval functions
         """
-        response = self._connection.make_request(
-            {},
+        response = self.connection.get(
             "modelci/eval_fn",
-            requests_command=requests.get,
         )
-        return [
-            EvalFunction(**eval_function)
-            for eval_function in response[EVAL_FUNCTIONS_KEY]
-        ]
-
-    def create_unit_test(self, name: str, slice_id: str) -> UnitTest:
+        payload = GetEvalFunctions.parse_obj(response)
+        return AvailableEvalFunctions(payload.eval_functions)
+
+    def create_unit_test(
+        self,
+        name: str,
+        slice_id: str,
+        evaluation_criteria: List[EvaluationCriteria],
+    ) -> UnitTest:
```

Comment:
As suggested by @pfmark, maybe we should force the list of `evaluation_criteria` to be nonempty

Reply:
I've added an empty check.